steps/expression.go

```diff
@@ -111,40 +128,115 @@ func (p *Expression) makeFunctions() map[string]govaluate.ExpressionFunction {
 		"set_timestamp": func(arguments ...interface{}) (interface{}, error) {
 			if len(arguments) == 1 {
-				if numArg, ok := arguments[0].(float64); ok {
-					p.currentSample().Time = time.Unix(int64(numArg), 0)
-					return arguments[0], nil
+				if numArg, err := p.argsToFloat64(arguments); err == nil {
+					outSampleAndHeader := bitflow.SampleAndHeader{
+						Sample: p.currentSample(),
+						Header: p.currentHeader(),
+					}
+					p.currentSample().Time = time.Unix(int64(numArg[0]), 0)
+					return outSampleAndHeader, nil
 				}
 			}
 			return nil, fmt.Errorf("set_timestamp() needs 1 float64 parameter, but received: %v", printParamStrings(arguments))
 		},
 		"floor": func(arguments ...interface{}) (interface{}, error) {
 			if len(arguments) == 1 {
-				if numArg, ok := arguments[0].(float64); ok {
-					return math.Floor(numArg), nil
+				if numArg, err := p.argsToFloat64(arguments); err == nil {
+					return math.Floor(numArg[0]), nil
 				}
 			}
 			return nil, fmt.Errorf("floor() needs 1 float64 parameter, but received: %v", printParamStrings(arguments))
 		},
+		"set": func(arguments ...interface{}) (interface{}, error) {
+			if len(arguments) != 2 && len(arguments) != 3 {
+				return nil, fmt.Errorf("following parametrization required: set(required(string), required(float64),"+
+					" optional(bitflow.SampleAndHeader)), but received: %v", printParamStrings(arguments))
+			}
+			var ok bool
+			outSampleAndHeader := bitflow.SampleAndHeader{
+				Sample: p.currentSample(),
+				Header: p.currentHeader(),
+			}
+			if len(arguments) == 3 {
+				outSampleAndHeader, ok = arguments[2].(bitflow.SampleAndHeader)
+				if !ok {
+					return nil, fmt.Errorf("Error while parsing third sample and header argument: %v", arguments[2])
+				}
+			}
+			value, ok := arguments[1].(float64)
+			if !ok {
+				return nil, fmt.Errorf("Error while parsing second float64 argument: %v", arguments[1])
+			}
+			field, ok := arguments[0].(string)
+			if !ok {
+				return nil, fmt.Errorf("Error while parsing first string argument: %v", arguments[0])
+			}
+			if ! outSampleAndHeader.Header.ContainsField(field) {
+				outSampleAndHeader.Sample = p.currentSample().DeepClone()
+				outSampleAndHeader.Header = p.currentHeader().Clone(p.currentHeader().Fields)
+
+				outSampleAndHeader.Sample.Values = append(outSampleAndHeader.Sample.Values, bitflow.Value(value))
+				outSampleAndHeader.Header.Fields = append(outSampleAndHeader.Header.Fields, field)
+			} else {
+				outSampleAndHeader.Sample.Values[p.varsToIndices[field]] = bitflow.Value(value)
+			}
+
+			return outSampleAndHeader, nil
+		},
+		"get_sample_and_header": func(arguments ...interface{}) (interface{}, error) {
+			outSampleAndHeader := bitflow.SampleAndHeader{
+				Sample: p.currentSample(),
+				Header: p.currentHeader(),
+			}
+			return outSampleAndHeader, nil
+		},
+	}
+}
+
+func (p *Expression) argsToFloat64(arguments interface{}) ([]float64, error) {
+	args, ok := arguments.([]interface{})
```

Comment:
Why not declare the `arguments` argument as `[]interface{}`, since that is always passed here anyways?